src/unix/pipe.c

```diff
@@ -85,6 +85,13 @@ int uv_pipe_bind2(uv_pipe_t* handle,
   if (flags & UV_PIPE_NO_TRUNCATE)
     if (namelen > sizeof(saddr.sun_path))
       return UV_EINVAL;
+  /*
+   * _PC_NAME_MAX returns the maximum length of a filename in the directory
+   * path that the process is allowed to create
+   * https://linux.die.net/man/3/pathconf
+   */
+  if (namelen > (size_t) pathconf(".", _PC_NAME_MAX))
```

Comment:
As pointed out by @saghul, this should be AIX-only. I don't know if AIX is susceptible but on other Unices this line fails to handle the following error cases:

1. The working directory may not exist or be inaccessible (i.e., pathconf can fail with ENOENT or EACCES)

2. The max may be different for "." and the file path of the unix socket (ex: nfs vs. local file system)